base/common/ReplxxLineReader.cpp

```diff
@@ -68,7 +91,20 @@ LineReader::InputStatus ReplxxLineReader::readOneLine(const String & prompt)
 
 void ReplxxLineReader::addToHistory(const String & line)
 {
+    // locking history file to prevent from inconsistent concurrent changes
+    bool locked = false;
+    if (flock(history_file_fd, LOCK_EX))
```

Comment:
Actually replxx already has file locking [1], but only for `history_save`.
Should the client be that safe? I.e. care about race during `history_load` (looks a little bit overkill for this)

[1]: https://github.com/AmokHuginnsson/replxx/blob/master/src/history.cxx#L123